src/cript/nodes/primary_nodes/project.py

```diff
@@ -76,6 +76,50 @@ def __init__(
         self._json_attrs = replace(self._json_attrs, name=name, collections=collections, material=material)
         self.validate()
 
+    def validate(self):
+        from cript.nodes.exceptions import (
+            CRIPTOrphanedMaterialError,
+            get_orphaned_experiment_exception,
+        )
+
+        # First validate like other nodes
+        super().validate()
+
+        # Check graph for orphaned nodes, that should be listed in project
+        # Project.materials should contain all material nodes
+        project_graph_materials = self.find_children({"node": ["Material"]})
+        # Combine all materials listed in the project inventories
+        project_inventory_materials = []
+        for inventory in self.find_children({"node": ["Inventory"]}):
+            for material in inventory.materials:
+                project_inventory_materials.append(material)
```

Comment:
this makes our code more implicit, do we want to make this more explicit?

I don't know enough on how to design this part, I think we need someone more senior than me who knows better

Reply:
I don't know.
Maybe. it's just a double list concat.